packages/logging-bunyan/src/index.js

```diff
@@ -0,0 +1,134 @@
+ * [Stackdriver Logging]{@link https://cloud.google.com/logging}.
+ *
+ * If your app is running on Google Cloud Platform, all configuration and
+ * authentication is handled for you. We also auto-detect the appropriate
+ * resource descriptor to report the log entries against.
+ *
+ * If you are running your application in anther environment, such as locally,
+ * on-premise, or on another cloud provider, you will need to provide additional
+ * configuration.
+ *
+ * @constructor
+ * @alias module:logging-bunyan
+ *
+ * @param {object} options - [Configuration object](#/docs). Refer to this link
+ *     for authentication information.
+ * @param {string=} options.logName - The name of the log that will receive
+ *     messages written to this bunyan stream. Default: `bunyan_Log`.
+ * @param {object=} options.resource - The monitored resource that the log
+ *     stream corresponds to. On Google Cloud Platform, this is detected
+ *     automatically, but you may optionally specify a specific monitored
+ *     resource. For more infromation see the
+ *     [official documentation]{@link
+ * https://cloud.google.com/logging/docs/api/reference/rest/v2/MonitoredResource}
+ *
+ * @example
+ * var bunyan = require('bunyan');
+ * var loggingBunyan = require('@google-cloud/logging-bunyan')({
+ *    projectId: 'grape-spaceship-123',
+ *    keyFilename: '/path/to/keyfile.json',
+ *    resource: {
+ *       type: 'global'
+ *    }
+ * });
+ *
+ * var logger = bunyan.createLogger({
+ *   name: 'my-service',
+ *   streams: [loggingBunyan.stream('info')]
+ * });
+ *
+ */
+function LoggingBunyan(options) {
+  if (!(this instanceof LoggingBunyan)) {
+    return new LoggingBunyan(options);
+  }
+
+  options = options || {};
+
+  this.logName_ = options.logName || 'bunyan_log';
+  this.resource_ = options.resource;
+  this.log_ = logging(options).log(this.logName_);
+}
+
+/**
+ * Convenience method that Builds a bunyan stream object that you can put in
+ * the bunyan streams list.
+ *
+ * @example
+ * var bunyan = reuqire('bunyan');
+ * var stackdriverBunyan = require('@google-cloud/logging-bunyan')(options);
+ * var logger = bunyan.createLogger({
+ *   name: 'my-service',
+ *   streams: [ stackdriverBunyan.stream('info') ]
+ * });
+ *
+ * @param {(string|number)} level - A bunyan logging level. Log entries at or
+ *     above this level would be routed to Stackdriver Logging.
+ */
+LoggingBunyan.prototype.stream = function(level) {
+  return {level: level, type: 'raw', stream: this};
+};
+
+/**
+ * Relay a log entry to the logging agent. This is normally called by bunyan.
+ *
+ * @param {object} rec - Bunyan log record.
+ * @private
+ */
+LoggingBunyan.prototype.write = function(rec) {
+  if (typeof rec === 'string') {
+    throw new Error(
+        '@google-cloud/logging-bunyan only works as a raw bunyan stream type.');
+  }
+  var level = BUNYAN_TO_STACKDRIVER[rec.level];
+  var entryMetadata = {resource: this.resource_, timestamp: rec.time};
```

Comment:
If the resource is null or undefined, @google-cloud/logging will try to autodetect the resource and apply it. The API rejects requests if they don't contain a valid resource.